packages/components/src/custom-select-control/test/index.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * External dependencies
+ */
+import { render, fireEvent } from '@testing-library/react';
+/**
+ * WordPress dependencies
+ */
+import { CustomSelectControl } from '@wordpress/components';
+
+describe( 'CustomSelectControl', () => {
+	const onKeyDown = jest.fn();
+
+	it( 'Captures the keypress event and does not let it propagate', () => {
+		const wrapper = (
+			<div
+				role="none"
+				className="test-class-name"
+				onKeyDown={ onKeyDown }
+			>
+				<CustomSelectControl options={ [ 'a', 'b', 'c' ] } />
+			</div>
+		);
+		const { container } = render( wrapper );
+		fireEvent.keyDown(
+			container.getElementsByClassName(
+				'components-custom-select-control__menu'
+			)[ 0 ]
+		);
```

Comment:
Using `getElementsByClassName` is not ideal, since we're [testing the implementation details](https://kentcdodds.com/blog/testing-implementation-details) of this component. Instead, we should write our tests from the point of view of the user, relying more on the accessible roles and the content that gets rendered — i.e:

1. open the select by clicking the toggle button
2. look for the element with role listbox (i.e. the equivalent of a `select`)
3. fire the event on that element

It is also a good practice to use the `screen` object instead of calling methods on the `container` (see [link](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-screen))

This would roughly translate to this code:

```js
import { render, fireEvent, screen } from '@testing-library/react';

// ...

const toggleButton = screen.getByRole( 'button' );
fireEvent.click( toggleButton );

const customSelect = screen.getByRole( 'listbox' );
fireEvent.keyDown( customSelectControl );
```

Reply:
Nice, thanks a lot for helping me out here! 👍🏼